src/api/IGDB_API/IGDB_API.tsx

```diff
@@ -0,0 +1,191 @@
+                'Authorization': this.userToken
+            },
+            data: `fields id, name, developed; where developed = (${data.map((game: Interfaces.IGDB_Game) => { return game.id })});`
+          })
+            .then((response: any) => {
+                developerData = response.data;         
+            })
+            .catch((err: any) => {
+                console.error(err);
+                throw new Error("Error fetching Game Developer data.")
+        });
+        return developerData;
+    }
+
+    private getGenres = async (data: Interfaces.IGDB_Game[]): Promise<Interfaces.IGDB_Genre[]> => {
+        let genreData: Interfaces.IGDB_Genre[] = [];
+        let genreIDs = data.map((game: Interfaces.IGDB_Game) => { return game?.genres?.[0] }).filter((element) => element !== undefined);
+
+        await axios({
+            url: " https://4fu7yxd9ml.execute-api.us-east-1.amazonaws.com/production/v4/genres",
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Authorization': this.userToken
+            },
+            data: `fields id, name; where id = (${genreIDs});`
+          })
+            .then((response: any) => {
+                genreData = response.data;         
+            })
+            .catch((err: any) => {
+                console.error(err);
+                throw new Error("Error fetching Genre data.")
+        });
+        return genreData;
+    }
+
+    private getImages = async (data: Interfaces.IGDB_Game[]): Promise<Interfaces.IGDB_Cover[]> => {
+        let coverData: Interfaces.IGDB_Cover[] = [];
+        let coverIDs = data.map((game: Interfaces.IGDB_Game) => { return game.cover }).filter((element) => element !== undefined);
+        let query = `fields *; where id = (${coverIDs});`;
+
+        await axios({
+            url: " https://4fu7yxd9ml.execute-api.us-east-1.amazonaws.com/production/v4/covers",
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Authorization': this.userToken
+            },
+            data: query
+          })
+            .then((response: any) => {
+                coverData = response.data;    
+            })
+            .catch((err: any) => {
+                console.error(err);
+                throw new Error("Error fetching Image data.")
+        });
+        return coverData;
+    }
+
+    // Formats IGDB Relational Data from API calls to game format
+        // -  Map platform IDs to platform names, and then separate games with multiple platforms into separate records
+        // -  Convert UNIX time to just the year
+        // -  Map developer IDs to developer names
+        // -  Map the first genre associated with the game to the genre name            
+    private formatData = (data: Interfaces.IGDB_Game[], platformData: Interfaces.IGDB_Platform[], developerData: Interfaces.IGDB_Developer[], genreData: Interfaces.IGDB_Genre[],
```

Comment:
I could see you eventually wanting to migrate this logic into its own API on the backend since its a bit complex and could get hefty if you wanted to display a long list of pages at once.